code/messaging/src/main/java/com/adobe/marketing/mobile/aepcomposeui/viewmodel/AepContainerViewModel.kt

```diff
@@ -0,0 +1,91 @@
+/*
+  Copyright 2025 Adobe. All rights reserved.
+  This file is licensed to you under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License. You may obtain a copy
+  of the License at http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software distributed under
+  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+  OF ANY KIND, either express or implied. See the License for the specific language
+  governing permissions and limitations under the License.
+*/
+
+package com.adobe.marketing.mobile.aepcomposeui.viewmodel
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import androidx.lifecycle.viewModelScope
+import com.adobe.marketing.mobile.aepcomposeui.AepContainerUI
+import com.adobe.marketing.mobile.aepcomposeui.AepUI
+import com.adobe.marketing.mobile.aepcomposeui.contentprovider.AepContainerUIContentProvider
+import com.adobe.marketing.mobile.aepcomposeui.contentprovider.AepUIContentProvider
+import com.adobe.marketing.mobile.aepcomposeui.utils.UIUtils
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.combine
+import kotlinx.coroutines.launch
+
+class AepContainerViewModel(
+    private val aepUIProvider: AepUIContentProvider,
+    private val aepContainerUIProvider: AepContainerUIContentProvider
+) : ViewModel() {
+
+    private val _uiList = MutableStateFlow(emptyList<AepUI<*, *>>())
+    val uiList: StateFlow<List<AepUI<*, *>>> = _uiList
+
+    private val _containerUI = MutableStateFlow<AepContainerUI<*, *>?>(null)
+    val containerUI: StateFlow<AepContainerUI<*, *>?> = _containerUI
+
+    private val _isLoaded = MutableStateFlow(false)
+    val isLoaded: StateFlow<Boolean> = _isLoaded
```

Comment:
Do you think grouping these values as a state holder like `AepContainerState` will be helpful for consumers?